src/controllers/user.controller.js

```
@@ -0,0 +1,99 @@
+import bcrypt from 'bcrypt';
+import { ApiError } from '../exceptions/api.error.js';
+import { jwtService } from '../services/jwt.service.js';
+import { tokenService } from '../services/token.service.js';
+import { userService } from '../services/user.service.js';
+import { emailService } from '../services/email.service.js';
+
+const getUserProfile = async (req, res) => {
+  const { refreshToken } = req.cookies;
+  const userData = jwtService.validateRefreshToken(refreshToken);
+
+  if (!userData) {
+    throw ApiError.Unauthorized();
+  }
+
+  const token = await tokenService.getByToken(refreshToken, 'refreshToken');
+
+  if (!token) {
+    throw ApiError.Unauthorized();
+  }
+
+  const userId = token.userId;
+
+  const user = await userService.getById(userId);
+
+  if (!user) {
+    throw ApiError.NotFound();
+  }
+
+  res.json(userService.normalize(user));
+};
+
+const updateUserName = async (req, res) => {
+  const { name } = req.body;
+
+  if (!name || name.trim() === '') {
+    return res.status(400).json({ message: 'Name is required' });
```

Instead of returning a 400 status directly, consider throwing an `ApiError.BadRequest` exception to maintain consistency with other error handling in the controller and allow the `errorMiddleware` to handle the response.